packages/block-editor/src/components/grid-visualizer/use-grid-layout-sync.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { GridRect } from './utils';
+
+export function useGridLayoutSync( { clientId: gridClientId } ) {
```

Comment:
Thinking out loud, it might make sense to trigger all of this logic from the control that toggles between Manual and Auto. That way we're not abusing `useEffect`.

Can change this later. Just something to think about

Reply:
Yeah that would be worth looking into!